code/game/objects/items/stacks/stack.dm

```diff
@@ -216,24 +216,29 @@
 	return get_amount() >= used
 
 /obj/item/stack/create_matter()
-	matter_per_piece = matter?.Copy() // this is used for refreshing matter amount in update_matter()
+	//Preserve matter_per_piece if already set, so it can be set manually.
+	if(!matter_per_piece)
```

Comment:
Can you give me an example of what you mean by inconsistencies?

Reply:
Why is that inconsistent? The matter list has no requirements to match exactly the amount * matter_per_piece in the base stack class. 
Some of the base stacks don't even have a matter cost (/obj/item/stack/medical, /obj/item/stack/telecrystal, some of /obj/item/stack/tile, /obj/item/stack/flag). 

But, you don't need that to tell the matter amount per unit. Because matter_per_piece is there for that? matter is the cost for the entire thing. Why couldn't it include some additional matter that's not affected by the amount? Especially in the /obj/item/stack class that has relaxed requirements on the matter calculation unlike /obj/item/stack/material?

Reply:
> Yes it does. This is literally the point of tracking matter per piece and updating the matter var. Stacks with no matter cost are irrelevant to that. The matter list is used for recycling the item in a lathe.
> 

So you're saying that since list/matter's use is for recycling the thing, it should not have a static amount of matter added to the matter amount calculated from matter_per_piece? 
I really don't know why you're so adamant about something like that? It sounds more like you're just taking a hard stance on that but don't want to tell me. If that's the case tell me so we can stop going back in forth and going nowhere.

> You are pulling this out of nowhere. Composite stacks exist.
>

You mean ``/obj/item/material`` stacks that can be reinforced. That concept doesn't exist at the ``/obj/item/stack`` level. Just go and read the code yourself? I'm not pulling anything out of nowhere?

Reply:
> I don't even know what you're trying to say here but I don't appreciate the tone.
> 

There's no "tone" here? What I was trying to say is if you got your idea already all made up, there's no point in debating it, and getting frustrated at each others. You should just say you don't want it and won't change your mind, so we can move onto something else.

> You are extrapolating the idea of matter purity or stringent requirements from code that doesn't have them.
>

I'm just saying that ``/obj/item/stack/material`` must have a matter list generated from its material and reinf_material, and won't in any way ever be compatible with having the matter list tampered with.

But the ``/obj/item/stack`` doesn't have to clear the whole thing because what it's made out of is less defined/automatic, or less implicit maybe? Basically, it comes down to generic vs specialized.

Pure matter count was referring to material stacks just automatically having their matter list filled by their material and reinforced material, while the base stack just makes a copy of the original matter list to be used as matter per piece.

> 'Go read the code yourself'; I wrote the code. Pull your head in.
>

I mean, you can have written the code a while ago and forgot what's written?
But I just looked at it again and "composite stacks" are only a thing at the /obj/item/stack/material level. Which is why I said you should look at the code, because you're telling me the opposite when I can see that's not the case. Its not an insult or anything?

Reply:
See, I understood "composite stack" as in when you reinforce material stacks, and end up with essentially a composite of two materials. Which is why I was insisting on that. There's no malice intended.